networkx/algorithms/flow/tests/test_networksimplex.py

```diff
@@ -385,3 +416,82 @@ def test_graphs_type_exceptions():
     pytest.raises(nx.NetworkXNotImplemented, nx.network_simplex, G)
     G = nx.DiGraph()
     pytest.raises(nx.NetworkXError, nx.network_simplex, G)
+
+
+def create_graph(
+    has_large_capacity: bool = False,
+    has_large_weight: bool = False,
+    has_large_demand: bool = False,
+):
+    large_value = 1000000000
+    G = nx.DiGraph()
+
+    # Add nodes with demands
+    if has_large_demand:
+        G.add_node("s0", demand=-large_value)
+        G.add_node("c1", demand=large_value)
+    else:
+        G.add_node("s0", demand=-4)
+        G.add_node("c1", demand=4)
+
+    G.add_node("s1", demand=-4)
+    G.add_node("ns", demand=0)
+    G.add_node("nc", demand=0)
+    G.add_node("c0", demand=4)
+
+    # Add edges
+    if has_large_capacity:
+        G.add_edge("s0", "ns", weight=1, capacity=large_value)
+    else:
+        G.add_edge("s0", "ns")
+
+    if has_large_weight:
+        G.add_edge("s1", "ns", weight=large_value)
+    else:
+        G.add_edge("s1", "ns", weight=1)
+
+    G.add_edge("ns", "nc", weight=1)
+    G.add_edge("nc", "c0", weight=1)
+    G.add_edge("nc", "c1", weight=1)
+
+    return G
+
+
+@pytest.mark.parametrize(
+    "has_large_capacity, has_large_weight, has_large_demand",
+    [
+        (True, True, True),
+        (True, True, False),
+        (True, False, True),
+        (True, False, False),
+        (False, True, True),
+        (False, True, False),
+        (False, False, True),
+        (False, False, False),
+    ],
+)
+def test_network_simplex_large_capacities(
+    has_large_capacity: bool, has_large_weight: bool, has_large_demand: bool
+):
+    """Address issues raised in ticket #7562."""
+    graph = create_graph(has_large_capacity, has_large_weight, has_large_demand)
+    flow_value, flow_dict = nx.network_simplex(graph)
+
+
+def test_network_simplex_unbounded_flow():
+    G = nx.DiGraph()
+    # Add nodes
+    G.add_node("A")
+    G.add_node("B")
+    G.add_node("C")
+
+    # Add edges forming a negative cycle
+    G.add_edge("A", "B", weight=-5)
+    G.add_edge("B", "C", weight=-5)
+    G.add_edge("C", "A", weight=-5)

```

Comment:
Just a cosmetic change, but this could be shortened:

```suggestion
    # A negative cycle
    G.add_weighted_edges_from([("A", "B", -5), ("B", "C", -5), ("C", "A", -5)])
```